### PR TITLE
Make `controller_addresses` and `address` required parameters

### PR DESCRIPTION
**Pull request by JayH5**
Defaulting to `$::ipaddress_lo` is not very useful.

### CODE REVIEW

**Comment by jerith** [comment]
Can we maybe have some tests that go beyond "it compiles"? :-)

**Comment by JayH5** [comment]
Yeah ok let me learn this stuff finally :p

File: spec/classes/controller_spec.rb [hunk 1]
```diff
@@ -4,10 +4,28 @@
   on_supported_os.each do |os, facts|
     context "on #{os}" do
       let(:facts) do
-        facts
+        facts.merge({:concat_basedir => '/tmp/puppetconcat'})
       end
 
-      it { should compile }
+      describe 'compiles when controller_addresses and address are specified' do
```
Comment (jerith) [inline spec/classes/controller_spec.rb]: Leave out the "compiles" here. That belongs to the `it { should compile }` below. The `describe` text should only contain the circumstances.

**Comment by jerith** [comment]
:+1:

**Pull request merged**